### DISCUSSION TITLE
Reactor y-scale ∝ maximum output?

### DISCUSSION THREAD

**Discussion by James-E-A** [FakeFishGames/Barotrauma#9073] [Ideas and suggestions]
I've got a [very small sub that has a 700kW reactor](https://steamcommunity.com/sharedfiles/filedetails/?id=2637468502); the fluctuations in power are tiny smears at the bottom of the display:

![602960_20220505190707_1](https://user-images.githubusercontent.com/2776014/167046143-c1bd4dd0-6ca7-469c-acc8-65414da08815.png)

Why isn't the y-axis scaled with the reactor's capacity?

I propose either of these suggestions:

* Set the top of the display to exactly 150% of the reactor's current/upgraded maximum output capacity (and the bottom to 0W)
* Allow the sub editor to set the value which the top of the display scales to

**Comment by Oukido** [comment]
That buggs me too.

The OP's solution is a **good** and simple approach. But batteries can still elevate the load graph line out of the bounds.
This could only be solved with dynamic scaling and shrinking tied to the highest value. (preferably with timed scaling steps and scaling indicators)

**Comment by Petethegoat** [comment]
**update:**
Here's the branch: https://github.com/Regalis11/Barotrauma/compare/master...Petethegoat:Barotrauma:reactorgraph-maxload

Tried implementing this. I think it's a good idea, and there's already code to raise the graph limit when the load is beyond the max output.

Graph on Barsuk in vanilla (10,000 KW max):
![image](https://user-images.githubusercontent.com/734762/197922010-488ba6ea-5716-4a24-b492-a8775198a8dc.png)

Graph on Barsuk topping off at 150% of max output (6300 KW max):
![image](https://user-images.githubusercontent.com/734762/197921545-7c9942e5-e0fd-46ca-a3cc-4f2403f42823.png)
The line position looks similar but note the slightly lower load.

And here's the graph with the reactor set to 1550KW max output (2325 KW max on graph):
![image](https://user-images.githubusercontent.com/734762/197927021-b45ceee0-241c-4739-8ae3-bcdf34bbecba.png)
